nucliadb/src/nucliadb/search/search/chat/ask.py

```diff
@@ -201,6 +196,23 @@ async def _stream(self) -> AsyncGenerator[AskResponseItemType, None]:
                 self.metrics.record_first_chunk_yielded()
                 first_chunk_yielded = True
 
+        # Then, rerank and cut (if needed) and stream out the retrieval results
+        if self._reranking is not None:
+            reranked = [
+                RankedItem(id=paragraph_id, score=score, score_type=SCORE_TYPE.RERANKER)
+                for paragraph_id, score in self._reranking.context_scores.items()
+            ]
+            apply_reranking(self.main_results, reranked)
+
+        yield RetrievalAskResponseItem(results=self.main_results)
+
+        if len(self.prequeries_results) > 0:
+            item = PrequeriesAskResponseItem()
+            for index, (prequery, result) in enumerate(self.prequeries_results):
```

Comment:
just a thought: should we rerank prequeries aswell?

Reply:
The prequeries are `FindRequest`s, so you can add a reranker option and will be reranked. However, they involve a round trip to predict, as we do them like a regular find